CMR-9260 creating the ingest and search draft concept API documents. (#1960)
nasa/Common-Metadata-Repository: schemas/resources/schemas/collection-draft/v1.0.0/search.md

### <a name="collection-draft"></a> Collection Draft

> Collection Drafts are draft records that describe a data set.Collection metadata is stored in the JSON format [UMM-Collection Schema](https://git.earthdata.nasa.gov/projects/EMFD/repos/unified-metadata-model/browse/collection).

Reviewer: nit;
```suggestion
Collection Drafts are draft records that describe a data set. Collection metadata is stored in the JSON format [UMM-Collection Schema](https://git.earthdata.nasa.gov/projects/EMFD/repos/unified-metadata-model/browse/collection).
```

| Field       | Description                                                        |
| ----------- | ------------------------------------------------------------------ |
| name        | the value of the Name field in Collection Draft metadata.      |
| id          | the CMR identifier for the result                                  |
| location    | the URL at which the full metadata for the result can be retrieved |
| revision-id | the internal CMR version number for the result                     |

__Example__

```
    curl -H "Cmr-Pretty: true" \
        "%CMR-ENDPOINT%/collection-drafts.xml?name=collection-name"
```

__Sample response__

```
    <?xml version="1.0" encoding="UTF-8"?>
    <results>
        <hits>1</hits>
        <took>13</took>
        <references>
            <reference>
                <name>collection-name</name>
                <id>CD1200000000-PROV1</id>
                <location>%CMR-ENDPOINT%/concepts/CD1200000000-PROV1/4</location>
                <revision-id>4</revision-id>
            </reference>
        </references>
    </results>
```

##### JSON

The JSON response includes the following fields.

> * hits - How many total Order Options were found.

Author: I thought I found them all, looks like I missed some. 
Thanks for finding them.
I fixed it.

* took - How long the search took in milliseconds

> * items - a list of the current page of Order Options with the following fields

Reviewer: ```suggestion
* items - a list of the current page of draft collections with the following fields
```

Author: Fixed it.

                },
                "umm": {
                    "ShortName": "Mapping Short Name 1.17.0 CMR-8220",
                    "Version": "1",
                    "EntryTitle": "Mapping Example for UMM-C 1.17.0 CMR-8220",
                    "DOI": {
                        "DOI": "10.1234/DOIID",
                        "Authority": "https://doi.org/"
                    },
                    ...
                    "MetadataSpecification": {
                        "URL": "https://cdn.earthdata.nasa.gov/umm/collection/v1.17.0",
                        "Name": "UMM-C",
                        "Version": "1.17.0"
                    }
                }
            }
        ]
    }
```

#### <a name="sorting-collection-draft-results"></a> Sorting Collection Draft Results

By default, Collection Draft results are sorted by name, then by provider-id.

One or more sort keys can be specified using the sort_key[] parameter. The order used impacts searching. Fields can be prepended with a - to sort in descending order. Ascending order is the default but + (Note: + must be URL encoded as %2B) can be used to explicitly request ascending.

> ##### Valid Order Option Sort Keys

Reviewer: ```suggestion
##### Valid Collection Draft Sort Keys
```
